File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.js
/**
 * @param {number[]} numbers
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(numbers, target) {
    // Init left and right as 2 pointers
    let left = 0;
    let right = numbers.length -1;
    // while left > right
    //  sum = numbers[left] + numbers[right]
    //  if sum equal to target
    //     we can return [left + 1,right + 1]
    //  else if sum < target
    //     we can increase left poitner by 1
    //  else sum > target
    //     we can decrease right pointer by 1
    
    while (left < right) {
        const sum = numbers[left] + numbers[right];
        
        if (sum === target) {
            return [left + 1, right + 1];
        } else if (sum < target) {
            left++;
        } else {
            right--;
        }
    }
    
    // Time complexity : O(n)
    // Space complexity : O(1)
    
};

File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        # Init left pointer = 0
        left = 0
        # Init right pointer = length of numbers - 1
        right = len(numbers) - 1
        
        # Init res array
        # res = []
        
        # while left < right:
        #   if left + right < target:
        #       increment left pointer
        #   else if left + right > target:
        #       decrease right pointer
        #   else:
        #       return [left + 1, right + 1]
        while left < right:
            if (numbers[left] + numbers[right]) < target:
                left += 1
            elif (numbers[left] + numbers[right]) > target:
                print(right)
                right -= 1
            else:
                return [left+1,right+1]
        
        # return res
        return []